### PR TITLE
Destination

### PR DESCRIPTION
**Pull request by maksenius**
### Description

add destination

### Quick checks:

- [x] I have followed the [Code Guidelines](https://github.com/ConduitIO/conduit/blob/main/docs/code_guidelines.md).
- [x] There is no other [pull request](https://github.com/ConduitIO/conduit-connector-snowflake/pulls) for the same update/change.
- [x] I have written unit tests.
- [x] I have made sure that the PR is of reasonable size and can be easily reviewed.

### CODE REVIEW

File: config/config_test.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+				},
+			},
+			want:    Config{},
+			wantErr: true,
+		},
+		{
+			name: "fail, missed table",
+			args: args{
+				cfg: map[string]string{
+					KeyConnection: "sqlserver://sa:password@0.0.0.0?database=mydb&connection+timeout=30",
```
Comment (maha-hajja) [inline config/config_test.go:63]: maybe we should have this keyConnection string as a constant for this file

File: destination/destination_test.go [hunk 2]
```diff
@@ -0,0 +1,273 @@
...
+	tests := []struct {
+		name    string
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "success",
+			args: args{
+				cfg: map[string]string{
+					config.KeyConnection: "sqlserver://sa:passwordx@0.0.0.0?database=mydb&connection+timeout=30",
```
Comment (maha-hajja) [inline destination/destination_test.go:48]: same for this string

File: destination/writer/writer.go [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+
+	"github.com/conduitio-labs/conduit-connector-sql-server/columntypes"
+)
+
+const (
+	// metadata related.
+	metadataTable = "sqlserver.table"
+)
+
+// Writer implements a writer logic for db2 destination.
```
Comment (maha-hajja) [inline destination/writer/writer.go:35]: db2 👀?

File: makefile [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+	go build -ldflags "-X 'github.com/conduitio-labs/conduit-connector-sql-server.version=${VERSION}'" -o conduit-connector-sql-server cmd/connector/main.go
+
+test:
+	go test $(GOTEST_FLAGS) ./...
+
+lint:
+	golangci-lint run
+
+mockgen:
+	mockgen -package mock -source destination/interface.go -destination destination/mock/destination.go
```
Comment (maha-hajja) [inline makefile:15]: can we also add these command to the file that has the interface? at the beginning of the file, like:
` //go:generate mockgen ...`

**Review by maha-hajja — COMMENTED** [review]
nice work! just added some small commnets

File: columntypes/columntypes.go [hunk 5]
```diff
@@ -0,0 +1,204 @@
...
+	"strings"
+	"time"
+
+	sdk "github.com/conduitio/conduit-connector-sdk"
+)
+
+const (
+	// sql server date, time column types.
+	dateType           = "date"
+	datetime2Type      = "datetime2Type"
```
[the remark this answers is not in the document]
Comment (maksenius) [reply columntypes/columntypes.go]: It is official sql server column type name
https://learn.microsoft.com/en-us/sql/t-sql/data-types/datetime2-transact-sql?view=sql-server-ver16

File: config/config_test.go (hunk 1 above)
Comment (maksenius) [reply config/config_test.go]: Good idea !
 done

File: destination/destination_test.go (hunk 2 above)
Comment (maksenius) [reply destination/destination_test.go:48]: fixed

File: makefile (hunk 4 above)
Comment (maksenius) [reply makefile:15]: I added comment how generate mock near interface declaration

File: columntypes/columntypes.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (maksenius) [reply columntypes/columntypes.go]: I have added comment and maybe we can change variable name `datetime2Type` to `datetime2ColumnType`. What do you think about it?

[the remark this answers is not in the document]
Comment (maha-hajja) [reply columntypes/columntypes.go]: no no let's keep it! didn't know about it before.. thanks for the explanation

File: makefile (hunk 4 above)
Comment (maha-hajja) [reply makefile:15]: I meant something like this https://github.com/ConduitIO/conduit/blob/e446e1c5e43319395151fdcc06b8726e0f7fcac5/pkg/orchestrator/orchestrator.go#L15 , we usually have the mock command near the interface, so when you change the interface, you would just click the run button near the `go:generate` command, and regenerate the mocks

Comment (maksenius) [reply makefile:15]: ok. I have done it

**Review by maha-hajja — APPROVED** [review]
LGTM!

**Pull request merged**